core/unix/os.c

```diff
@@ -2454,12 +2518,19 @@ os_thread_init(dcontext_t *dcontext, void *os_data)
     }
 #endif
 
+    /* FIXME i#3544: RFC: this log essentially prints DR TLS registers but
```

Comment:
By "keeping order consistent" you mean between the two log statements?

The current modifications you made look fine. Not sure what the suggestion meant to improve?

Reply:
> By "keeping order consistent" you mean between the two log statements?

Yes. Current code will always print `gs` first and `fs` second. If that is not required, then in `os_exports.h` we could have:
```c
#ifdef X86
#    if defined(MACOS64)
// ...
#        define STR_SEG "gs"
#        define STR_LIB_SEG "fs"
#    elif defined(X64)
// ...
#        define STR_SEG "gs"
#        define STR_LIB_SEG "fs"
#    else
// ...
#        define STR_SEG "fs"
#        define STR_LIB_SEG "gs"
#    endif
#elif defined(AARCHXX)
#    ifdef X64
// ...
#        define STR_SEG "tpidruro"
#        define STR_LIB_SEG "tpidrurw"
#    else
// ...
#        define STR_SEG "tpidrurw"
#        define STR_LIB_SEG "tpidruro"
#    endif                                  /* 64/32-bit */
#elif defined(RISCV64)
// ...
#    define STR_SEG "tp"
#    define STR_LIB_SEG "s11"
#endif /* X86/ARM */
```
And in `os.c` this logs would be:
```c
    LOG(THREAD, LOG_THREADS, 1, "post-TLS-setup, cur %s base is " PFX "\n",
        STR_SEG, get_segment_base(SEG_TLS));
    LOG(THREAD, LOG_THREADS, 1, "post-TLS-setup, cur %s base is " PFX "\n",
        STR_LIB_SEG, get_segment_base(LIB_SEG_TLS));
```

Reply:
Sounds good in general. See comments though.

Reply:
Done.